### PR TITLE
NEP-6 wallet support

### PR DESCRIPTION
**Pull request by slipo**
**What current issue(s) from Trello/Github does this address?**
https://github.com/CityOfZion/neon-wallet/issues/421

**What problem does this PR solve?**
All projects should follow the NEP-6 wallet standard so they can share wallets

**How did you solve this problem?**
Some notes:
I leave the keys.json storage file behind, just in case.

When we upgrade the wallets we only have labels and encrypted keys. This means that we can’t fill in the address fields until they log into the accounts.

I first tried to use neon-js which was pretty cool but it throws an exception if you try to export an Account without an address. It wasn’t too much work to do it all custom in the end.

Try to use naming inline with the NEP-6 standard (wallets contain accounts, in settings you add accounts not keys)

**How did you make sure your solution works?**
Automated + manual tests

QA Tests worth doing (any others?)
 * Starting with legacy key file, it should migrate to NEP-6. Delete an account, restart app and make sure that account is still missing.
 * Starting with no wallet, test recovering NEP-6 wallet
 * Starting with no wallet, test recovering legacy wallet
 * Adding accounts after the two above
 * Starting with no wallet, test adding 2 accounts, logging into both works 
 * Recovering NEP-6 / Legacy after already having accounts in the wallet
 * Logging into recovered legacy account
 * Logging into recovered NEP-6 account
 * Deleting all those accounts
 * Recovering same recovery file multiple times (won’t allow key to be in wallet twice)
 * Export everything, delete all accounts, import

**Are there any special changes in the code that we should be aware of?**
No.

**Is there anything else we should know?**
Having wallet accounts without public addresses could pose a challenge if other programs the user is importing into don't support that. There's not really anything we can do though.

- [x] Unit tests written?

### CODE REVIEW

**Comment by slipo** [comment]
Per this issue:
https://github.com/electron-userland/electron-builder/issues/2260

I upgraded electron-builder and now the build works.

File: app/containers/Settings/Settings.jsx [hunk 1]
```diff
@@ -102,28 +105,33 @@ export default class Settings extends Component<Props, State> {
...
-      text: `Please confirm deleting saved wallet - ${key}`,
+      text: `Please confirm deleting saved wallet - ${label}`,
       onClick: () => {
         // eslint-disable-next-line
-        storage.get('keys', (error, data) => {
-          delete data[key]
-          storage.set('keys', data)
-          setKeys(data)
+        storage.get('userWallet', (error, data) => {
+          data.accounts.map((account, index) => {
```
Comment (evgenyboxer) [inline app/containers/Settings/Settings.jsx]: Can we do some error handling here?
` if (error) { ... } else { delete wallet account }`
Also can we use something like:
`const keys = reject(data.accounts, { key });`
Also, why do we need to check the label when deleting? isn't a key is unique?

File: __tests__/modules/generateWallet.test.js [hunk 2]
```diff
@@ -29,31 +35,183 @@ describe('generateWallet module tests', () => {
 
   describe('newWallet tests', () => {
     const payload = account
-    const expectedAction = Object.assign({}, { payload }, { type: NEW_WALLET })
+    const expectedAction = Object.assign({}, { payload }, { type: NEW_WALLET_ACCOUNT })
```
Comment (evgenyboxer) [inline __tests__/modules/generateWallet.test.js]: Do we need an Object.assign here?
Can it be - 
`const expectedAction = { payload, type: NEW_WALLET_ACCOUNT }`

File: app/modules/generateWallet.js [hunk 3]
```diff
@@ -27,22 +28,152 @@ export function newWallet (account: Object) {
...
+    label: label,
+    isDefault: isDefault, // Make the first account the default
+    lock: false,
+    key: key,
+    contract: {},
+    extra: null
+  }
+}
+
+export const createEmptyWallet = () => {
```
Comment (evgenyboxer) [inline app/modules/generateWallet.js]: How about this - 
```
export const createEmptyWallet = ({
  ...DEFAULT_WALLET,
  scrypt: DEFAULT_SCRYPT
})
```

File: app/modules/generateWallet.js [hunk 4]
```diff
@@ -27,22 +28,152 @@ export function newWallet (account: Object) {
...
+        resolve()
+      }
+    })
+  })
+}
+
+export const walletHasKey = (wallet, key) => {
+  let foundKey = false
+
+  each(wallet.accounts, (account) => {
```
Comment (evgenyboxer) [inline app/modules/generateWallet.js]: Use .some - https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some

`wallet.accounts.some((account) => account.key === key)`

**Pull request merged**